packages/entity-browser/src/modules/entityBrowser/reducer.spec.js

```diff
@@ -0,0 +1,238 @@
+              records: {
+                data: 'data'
+              }
+            }
+          }
+
+          const expectedStateAfter = {
+            recordStore: {}
+          }
+
+          expect(reducer(stateBefore, actions.clearRecordStore())).to.deep.equal(expectedStateAfter)
+        })
+
+        it('should handle ADD_RECORDS_TO_STORE', () => {
+          const existingRecords = {
+            '1': [
+              {field: 'value1'},
+              {field: 'value2'}
+            ]
+          }
+          const newRecords = [
+            {field: 'value3'},
+            {field: 'value4'}
+          ]
+
+          const mergedRecords = {
+            '1': [
+              {field: 'value1'},
+              {field: 'value2'}
+            ],
+            '2': [
+              {field: 'value3'},
+              {field: 'value4'}
+            ]
+          }
+
+          const stateBefore = {
+            recordStore: existingRecords
+          }
+
+          const expectedStateAfter = {
+            recordStore: mergedRecords
+          }
+
+          expect(reducer(stateBefore, actions.addRecordsToStore(2, newRecords))).to.deep.equal(expectedStateAfter)
+        })
+
+        it('should handle SET_ENTITY_NAME', () => {
+          const stateBefore = {
+            entityName: 'EntityBefore'
+          }
+
+          const expectedStateAfter = {
+            entityName: 'EntityAfter'
+          }
+
+          expect(reducer(stateBefore, actions.setEntityName('EntityAfter'))).to.deep.equal(expectedStateAfter)
+        })
+
+        it('should handle SET_RECORDS', () => {
+          const newRecords = [
+            {
+              lastname: 'Lastname1',
+              firstname: 'Firstname1'
+            },
+            {
+              lastname: 'Lastname2',
+              firstname: 'Firstname2'
+            }
+          ]
+
+          const stateBefore = {
+            records: [
+              {
+                lastname: 'LastnameBefore',
+                firstname: 'FirstnameBefore'
+              }
+            ]
+          }
+
+          const expectedStateAfter = {
+            records: newRecords
+          }
+
+          expect(reducer(stateBefore, actions.setRecords(newRecords))).to.deep.equal(expectedStateAfter)
+        })
+
+        it('should handle SET_COLUMN_DEFINITION', () => {
+          const newDefinition = [
+            {
+              label: 'col1',
+              value: [{0: 'col1'}]
+            }, {
+              label: 'col2',
+              value: [{0: 'col2'}]
+            }
+          ]
+
+          const stateBefore = {
+            columnDefinition: []
+          }
+
+          const expectedStateAfter = {
+            columnDefinition: newDefinition
+          }
+
+          expect(reducer(stateBefore, actions.setColumnDefinition(newDefinition))).to.deep.equal(expectedStateAfter)
+        })
+
+        it('should handle SET_SEARCH_TERM', () => {
+          const newSearchTerm = '_search=New_search_term'
```

Comment:
is this test "wrong" or does the searchterm gets persisted like this? (with _search=)